tests/rainbowGrades/test_csci1100.py

```diff
@@ -0,0 +1,295 @@
+
+    print("Checking summary files against expected summaries.")
+    # Verify that a valid copy of output.html was sent to all_students_summary_html
+    make_output = make_output.splitlines()
+    make_output = make_output[-1].strip()  # Get the RUN COMMAND LINE
+    make_output = make_output.split(b'/')
+    make_output = make_output[-1]  # Get the name of the output.html file since it uses the date
+    if not os.path.exists(os.path.join(summary_tmp, "all_students_summary_html", make_output)):
+        error_and_cleanup(test_tmp, "Failed to find output file in all_students_summary_html")
+
+    output_generated_contents = ""
+    output_known_contents = ""
+    try:
+        with open(os.path.join(summary_tmp, "output.html"), 'r') as output_generated_file,\
+             open(os.path.join(script_path, "output_10090542_csci1100.html"), 'r') as output_known_file:
+            output_generated_contents = output_generated_file.read()
+            output_known_contents = output_known_file.read()
+    except Exception as e:
+        error_and_cleanup(test_tmp, "{}".format(e))
+
+    if output_generated_contents != output_known_contents:
+        error_and_cleanup(test_tmp, "Generated output.html did not match expected output.html")
+
+    # Extract test version of individual_grade_summary_html files
+    known_individual_path = os.path.join(test_tmp, "individual_summary_html")
+    summary_individual_path = os.path.join(summary_tmp, "individual_summary_html")
+    os.mkdir(known_individual_path)
+    return_code = subprocess.call(["tar", "-xf", os.path.join(script_path, "individual_summary_10090542_csci1100.tar"),
+                     "-C", known_individual_path])
+    if return_code != 0:
+        error_and_cleanup(test_tmp, "Extracting raw data failed (Error {}".format(return_code))
+
+    # Construct lists of generated and test individual_grade_summary_html files
+    known_files = []
+    summary_files = []
+    for f in os.listdir(known_individual_path):
+        known_files.append(f)
+    for f in os.listdir(summary_individual_path):
+        summary_files.append(f)
+
+    # Check that the name and number of individual_grade_summary_html files are the same
+    if len(known_files) != len(summary_files):
+        file_diff = len(known_files) - len(summary_files)
+        if len(known_files) > len(summary_files):
+            error_and_cleanup(test_tmp, "There are {} more files in the generated ".format(file_diff) +
+                              "individual_summary_html than expected.")
+        else:
+            error_and_cleanup(test_tmp, "There are {} fewer files in the generated ".format(-1*file_diff) +
+                              "individual_summary_html than expected.")
+
+    # Compare the contents (excluding time-sensitive parts) of generated and test individual_grade_summary_html files
+    # TODO: Currently not checking generated personal message files (for seating/materials)
+    for f in known_files:
```

Comment:
There's a whole bunch of _summary.html files - code above it checks to make sure that the names and number of files in the directory matches the expected directory so we can't enter this loop without being guaranteed these files are named correctly and exist.